File: src/go-backend/internal/models/service.go
package models

// Service Response
type Service struct {
	BackdropUrl        string   `json:"backdropUrl"`
	ID                 string   `json:"id"`
	Name               string   `json:"name"`
	OriginalName       string   `json:"originalName"`
	Overview           string   `json:"overview"`
	PosterUrl          string   `json:"posterUrl"`
	MediaTypes         []string `json:"mediaTypes"`
	OriginalLanguage   string   `json:"originalLanguage"`
	HomePage           string   `json:"homePage"`
	OriginCountry      string   `json:"originCountry"`
	AvailabilityStatus string   `json:"availabilityStatus"`
}

// Services Response
type Services struct {
	ServiceCount int        `json:"serviceCount"`
	Services     []*Service `json:"services"`
}

File: src/frontend/src/lib/types/service.ts
export type ServiceRequest = {
    ServiceTag: string;
};

export type Service = {
    backdropURL: string;
    id: string;
    name: string;
    originalName: string;
    overview: string;
    posterURL: string;
    mediaTypes: string[];
    originalLanguage: string;
    homePage: string;
    originCountry: string;
    availabilityStatus: string;
};

export type Services = {
    serviceCount: number;
    services: Service[];
};
